store/kv/kv.go

```diff
@@ -76,6 +85,103 @@ func NewKeyValueStore(cfg *config.FoundationDBConfig) (KeyValueStore, error) {
 	}, nil
 }
 
+func NewKeyValueStoreWithMetrics(cfg *config.FoundationDBConfig) (KeyValueStore, error) {
+	kv, err := newFoundationDB(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &KeyValueStoreImplWithMetrics{
+		&KeyValueStoreImpl{
+			fdbkv: kv,
+		},
+	}, nil
+}
+
+func (m *KeyValueStoreImplWithMetrics) measure(name string, f func() error) {
+	tags := metrics.GetFdbReqTags(name, false)
+	defer metrics.FdbRequests.Tagged(tags).Histogram("histogram", tally.DefaultBuckets).Start().Stop()
+	var fdberr fdb.Error
+	err := f()
```

Comment:
Maybe reorder to reduce branching for readability
```
err := f()
if err == nil {
	metrics.FdbRequests.Tagged(tags).Counter("ok").Inc(1)
    return
}

var fdberr fdb.Error
if !errors.As(err, &fdberr) {
    metrics.FdbErrorRequests.Tagged(tags).Counter("unknown").Inc(1)
    return
}
		
errTags := metrics.GetFdbReqSpecificErrorTags(name, strconv.Itoa(fdberr.Code), false)
metrics.FdbErrorRequests.Tagged(errTags).Counter("specific").Inc(1)
	
```

Reply:
Good catch!